Assets/Scripts/AI/MapEntry.cs

```diff
@@ -0,0 +1,27 @@
+using System.Collections.Generic;
+using System.Linq;
+
+namespace AI
+{
+    public class MapEntry
+    {
+        private readonly SortedDictionary<string, float> _attributes;
+
+        public MapEntry(SortedDictionary<string, float> attributes)
+        {
+            _attributes = attributes;
+        }
+
+        /// <summary>
+        /// The wrap float attributes of an object.
+        /// </summary>
+
+        public List<float> Attributes => _attributes != null ? _attributes.Values.ToList() : new List<float>();
```

Comment:
you are recalculating the list every time you query it or its size. 

maybe just use float[] and ditch the SortedDict for memory and performance efficiency

only calculate the float[] once in the AiWrapper and then save it in the MapEntry (which technically might not be necessary you could use a [x,y][dim] here if that works?)

Reply:
As you mentioned in the chat, I think the sorting is necessary so every time the attributes are in the correct dimension. But the dimension calculation performance can be increased.